compiler/lib/src/main/scala/codegen/CppWriter/ComponentCppWriter/ComponentCppWriter.scala

```diff
@@ -0,0 +1,973 @@
+          CppDoc.Type("NATIVE_INT_TYPE"),
+          "msgSize",
+          Some("The message size")
+        )
+      )
+      else Nil
+
+    addAccessTagAndComment(
+      "public",
+      "Component initialization",
+      List(
+        functionClassMember(
+          Some(s"Initialize $className object"),
+          "init",
+          initQueueDepthParam ++ initMsgSizeParam ++ initInstanceParam,
+          CppDoc.Type("void"),
+          intersperseBlankLines(
+            List(
+              lines(
+                s"""|// Initialize base class
+                    |Fw::$baseClassName::init(instance);
+                    |"""
+              ),
+              intersperseBlankLines(
+                List(
+                  intersperseBlankLines(specialInputPorts.map(writePortConnections)),
+                  intersperseBlankLines(typedInputPorts.map(writePortConnections)),
+                  intersperseBlankLines(serialInputPorts.map(writePortConnections)),
+                  intersperseBlankLines(specialOutputPorts.map(writePortConnections)),
+                  intersperseBlankLines(typedOutputPorts.map(writePortConnections)),
+                  intersperseBlankLines(serialOutputPorts.map(writePortConnections)),
+                  data.kind match {
+                    case Ast.ComponentKind.Passive => Nil
+                    case _ => List.concat(
+                      if hasSerialAsyncInputPorts then lines(
+                        """|// Passed-in size added to port number and message type enumeration sizes.
+                           |// NATIVE_INT_TYPE cast because of compiler warning.
+                           |this->m_msgSize = FW_MAX(
+                           |  msgSize +
+                           |  static_cast<NATIVE_INT_TYPE>(sizeof(NATIVE_INT_TYPE)) +
+                           |  static_cast<NATIVE_INT_TYPE>(sizeof(I32)),
+                           |  static_cast<NATIVE_INT_TYPE>(ComponentIpcSerializableBuffer::SERIALIZATION_SIZE)
+                           |);
+                           |
+                           |Os::Queue::QueueStatus qStat = this->createQueue(queueDepth, this->m_msgSize);
+                           |"""
+                      )
+                      else lines(
+                        """|Os::Queue::QueueStatus qStat = this->createQueue(
+                           |  queueDepth,
+                           |  ComponentIpcSerializableBuffer::SERIALIZATION_SIZE
+                           |);
+                           |"""
+                      ),
+                      lines(
+                        """|FW_ASSERT(
+                           |  Os::Queue::QUEUE_OK == qStat,
+                           |  static_cast<FwAssertArgType>(qStat)
+                           |);
+                           |"""
+                      )
+                    )
+                  }
+                )
+              )
+            )
+          )
+        )
+      )
+    )
```

Comment:
This is a lot of close parentheses in a row. Consider breaking this up by assigning some of the inner expressions to variables.

Reply:
Ok, fixed!

Reply:
Looks great!